docs/how-to/internal/release.rst

```diff
@@ -0,0 +1,72 @@
+============================
+How to cut a Rubicon release
+============================
+
+The release infrastructure for Rubicon is semi-automated, using Github
+Actions to formally publish releases.
+
+This guide assumes that you have an ``upstream`` remote configured on your
+local clone of the Rubicon repository, pointing at the official repository.
+It also assumes that you have a local ``release`` branch that tracks
+``upstream/master``.
+
+You should have a checkout of a personal fork of the Rubicon repository; to
+configure this fork for a release, run::
+
+    $ git remote add upstream git@github.com:beeware/rubicon-objc.git
+    $ git clone upstream/master release
```

Comment:
So - this is something I do locally, but I'm open to suggestions on other approaches.

The "release" branch on my local checkouts isn't a "release" branch - it's the master on upstream (and you're right - that *clone* syntax isn't legal; it should be `checkout -b`, and the `--track` addition is an option I've got turned on globally, but isn't necessarily set like that). 

I can't release from "master", because master is origin/master, not upstream/master. I guess I could upstream/master as my local branch name instead of release...

Reply:
Clearly our forks and local repos aren't set up the same way. 😄 I take it that your `master`/`origin/master` points to something other than `upstream/master`?

In my case, my local repo has `origin` and `beeware` (≈ `upstream`) remotes. I have a `master` branch locally, which I always keep up-to-date with `beeware/master`, and regularly push to `origin` (i. e. my GitHub fork). I use this `master` branch when I want to get my local copy to the same state as upstream, and as the starting point for new PR branches.

In any case, perhaps we should leave this unchanged for now to be safe. (As long as rubicon-objc doesn't have an upstream `release` branch there shouldn't be any conflicts.)

Reply:
I generally have origin pointing to my own fork, and upstream pointing to the "original" repo, with "master" being "origin/master" (which sounds like it's the same as your setup?)

So, it turns out the "release" branch is only needed if you're planning to commit directly to upstream/master - which we shouldn't be doing for Rubicon (I do on a couple of other beeware projects because we haven't quite got to the critical mass of reliably having non-me reviewers).

Regardless, I'm definitely not planning to introduce an upstream/release branch any time soon.